### ISSUE TITLE
Hacking the "Mini LED 9.92" software bundled with so-called 2017 version of b1248 (actually a GD1144)?

### ISSUE THREAD

**Issue by Danimations** [Effix/LedBadge#11]
Hi again, @Effix . 

Today I tried working with the extreme constraints of the software that came bundled with my 2017 version of the b1248 LED badge (actually a GD1144 mislabelled). 

Using the Windows program "Mini LED 9.92" I am able to load up to 8 animation frames onto the badge... but the playback is interrupted with blank frames. After each animation frame, the screen dips to black then up again with the next frame. The resulting effect is intense, distracting, flashing.

With the device set to the highest playback speed (8 on a scale of 1 to 8) I get a playback speed roughly 6-10 fps. This is limited, but workable... but for the flashing. The flashing makes it painful to look at.

I wonder if it would be possible to modify the existing Mini LED 9.92 loader software to eliminate the flashing problem? That would make the newer badges usable for simple animations of up to 8 frames in length.

The software's About menu claims (c) 2015, but doesn't name an author or responsible company.

I realise that this would be an entirely separate project, but thought I'd raise the possibility with you given your interest in these badges.

**Comment by Effix** [comment]
I don't think it would be possible to change the badge behavior by hacking at the computer software. You couldn't add anything that wasn't already supported by the code on the badge, anyway.

Assuming the badge actually did what you wanted it to as is, using/modifying one of the python utilities that others have written would be easier than trying to modify the program that came with the badge.

**Comment by Danimations** [comment]
Well, the problem I have at the moment is that the software loader wasn't obviously designed to facilitate _animation_ as such. It really just makes static images scroll or wipe.

That said, when I flick the power switch to "off", the badge begins to charge and a six-frame animation plays. It shows a "battery charging" animated icon while a two-frame loop of a steaming coffee cup repeats.

This is smooth (albeit low frame rate) animation, without the blanking between frames that I've encountered using the Windows loader. The coffee cup/power bar uninterrupted animation is what I'm looking to achieve, but I can't obviously achieve it with the Windows software.

I should be able to use your firmware, but since I appear to have incompatible 2017 hardware, that's currently a non-starter for me. I may well have to find another way to my destination.

What would you do if you were in my position?
